File: src/main/java/com/xeoneux/caravansary/converter/ReservationRequestToReservationEntityConverter.java
package com.xeoneux.caravansary.converter;

import com.xeoneux.caravansary.entity.ReservationEntity;
import com.xeoneux.caravansary.request.ReservationRequest;
import org.springframework.core.convert.converter.Converter;

public class ReservationRequestToReservationEntityConverter
        implements Converter<ReservationRequest, ReservationEntity> {
    @Override
    public ReservationEntity convert(ReservationRequest source) {
        ReservationEntity reservationEntity = new ReservationEntity();
        reservationEntity.setCheckin(source.getCheckin());
        reservationEntity.setCheckout(source.getCheckout());

        if (source.getId() != null) reservationEntity.setId(source.getId());

        return reservationEntity;
    }
}

File: src/main/kotlin/com/xeoneux/caravansary/converter/ReservationRequestToReservationEntityConverter.kt
package com.xeoneux.caravansary.converter

import com.xeoneux.caravansary.entity.ReservationEntity
import com.xeoneux.caravansary.request.ReservationRequest
import org.springframework.core.convert.converter.Converter

class ReservationRequestToReservationEntityConverter : Converter<ReservationRequest, ReservationEntity> {
    override fun convert(source: ReservationRequest): ReservationEntity {
        val reservationEntity = ReservationEntity()
        reservationEntity.checkin = source.checkin
        reservationEntity.checkout = source.checkout

        if (source.id != null) reservationEntity.id = source.id

        return reservationEntity
    }
}
